circuit/environment/src/helpers/variable.rs

```diff
@@ -90,6 +90,19 @@ impl<F: PrimeField> Variable<F> {
             }
         }
     }
+
+    ///
+    /// Returns the relative index and value of the variable.
+    ///
```

Comment:
Might be nice to document as reminder to readers that `Public` variables occupy indices [0, m] for some m, and `Private` will then occupy [m + 1, n].

Reply:
Don't they have their individual indices?
```
    pub(crate) fn new_public(&mut self, value: F) -> Variable<F> {
        let variable = Variable::Public(Arc::new((self.public.len() as u64, value)));
        self.public.push(variable.clone());
        self.counter.increment_public();
        self.num_variables += 1;
        variable
    }

    pub(crate) fn new_private(&mut self, value: F) -> Variable<F> {
        let variable = Variable::Private(Arc::new((self.private.len() as u64, value)));
        self.private.push(variable.clone());
        self.counter.increment_private();
        self.num_variables += 1;
        variable
    }

```